### DISCUSSION TITLE
Should i be concerned by performance when calling hasRole or hasPermissionTo

### DISCUSSION THREAD

**Discussion by grunk** [spatie/laravel-permission#2066] [Q&A]
Hi,

maybe it's a dumb question but i come from a "low level" framework where everything was very clear about when a db query is made or not.
With laravel i'm kind of lost and wondering if i should be careful how often i call `$user->hasRoles('xxxx')` or `$user->hasPermissionTo('xxxx')`.

Let's say i'm calling 10x `auth()->user()->hasRoles()`, will this translate to a new query each time because everything is lazy loaded or the current user already have all this roles and permissions loaded up at bootstrap ans i can call the functions as many times as I want ?

Thanks

**Comment by erikn69** [comment]
> will this translate to a new query each time

Not, only one query
